lib/rubocop/cop/style/class_and_module_children.rb

```diff
@@ -140,6 +161,71 @@ def check_nested_style(node)
           end
         end
 
+        def check_namespaced_style(node)
+          return if already_warned_about_class_as_namespace?(node)
+
+          if LowestLevelConstantDefinition.check(node)
+            check_class_or_module_definition_namespacing(node)
+          else
+            check_namespacing_node?(node)
+          end
+        end
+
+        def check_namespacing_node?(node)
+          return if class_as_namespace(node)
+          return if NamespacingModule.check(node)
+          return if LowestLevelConstantDefinition.check(node)
+          return if already_warned_about_namespace_in_compact_style?(node)
+
+          warn_about_namespace_in_compact_style(node)
+          add_offense(node.loc.name, message: NAMESPACING_MSG) do |corrector|
+            namespace_definition(corrector, node)
+          end
+
+          nil
+        end
+
+        def class_as_namespace(node)
+          return false unless UsingClassAsNamespace.check(node)
+
+          warn_about_class_as_namespace(node)
+          add_offense(node.loc.name, message: USING_CLASS_AS_NAMESPACE_MSG)
+
+          true
+        end
+
+        def warn_about_namespace_in_compact_style(node)
+          already_warned_about_namespace_in_compact_style.add node
+        end
+
+        def already_warned_about_namespace_in_compact_style?(node)
+          node.ancestors.any? { |n| already_warned_about_namespace_in_compact_style.include?(n) }
+        end
+
+        def already_warned_about_namespace_in_compact_style
+          @already_warned_about_namespace_in_compact_style ||= Set.new
```

Comment:
I am not sure if this approach is necessary. What I found is that I could get offenses to render correctly with nested modules but the auto-correct was clobbering the tree. 

I think warning once on the outer namespace could probably get the point across.